src/utils/sformat.fs

```diff
@@ -54,65 +54,73 @@ namespace Microsoft.FSharp.Text.StructuredFormat
 
     [<NoEquality; NoComparison>]
 #if COMPILER
+    type internal BoxRange = BoxRange of obj
```

Comment:
I guess this is fixable, by moving range file on it's own (if it doesn't contain too much stuff as well), maybe try a PR on this PR which just moves the files in the relevant fsprojs if that's doable.

I think the hack is fine (looking at `unboxRange` and its usage) but it needs an explicit comment where the type is defined as well as in `unboxRange`, and having a statement (beside the picture from @vasily-kirichenko) that this move was attempted but seemed disruptive would be good too.